src/java/org/apache/cassandra/db/rows/PartitionSerializationException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.rows;
+
+import org.apache.cassandra.schema.TableMetadata;
+
+public class PartitionSerializationException extends RuntimeException
+{
+    public PartitionSerializationException(UnfilteredRowIterator partition, Throwable cause)
+    {
+        super(buildMessage(partition), cause);
+    }
+
+    private static String buildMessage(UnfilteredRowIterator partition)
+    {
+        TableMetadata table = partition.metadata();
+        Object readablePartitionkey = table.partitionKeyType.compose(partition.partitionKey().getKey());
```

Comment:
Looks like `partitionKeyType.getString()` is more commonly used in the code base to get the readable representation of partition key. 

Btw, both `getString` and `compose and toString` produce the same result. 
But for the **composite partition key**, the output is not so nice. It contains utf8 characters. For the table 
```
CREATE TABLE ks.tbl (key ascii, name ascii, val ascii, val1 ascii, PRIMARY KEY ((key, name), val))
```
The partition key string for `key1, name1` is 
```
\x00\x04key1\x00\x00\x04name1\x00
```

Reply:
It would be nice to have a test to verify if the message matches expectation.

Reply:
> But for the composite partition key, the output is not so nice. It contains utf8 characters.

Looks like switching to `getString()` at minimum makes sense. I'm not sure if we should solve the composite problem. Will noodle on it...

> a test to verify if the message matches expectation

Agreed.